python/lsst/sims/catUtils/mixins/VariabilityMixin.py

```diff
@@ -59,49 +128,128 @@ def initializeVariability(self, doCache=False):
 
 
 
-    def applyVariability(self, varParams):
+    def applyVariability(self, varParams_arr):
+        """
+        Read in an array/list of varParamStr objects taken from the CatSim
+        database.  For each varParamStr, call the appropriate variability
+        model to calculate magnitude offsets that need to be applied to
+        the corresponding astrophysical offsets.  Return a 2-D numpy
+        array of magnitude offsets in which each row is an LSST band
+        in ugrizy order and each column is an astrophysical object from
+        the CatSim database.
         """
-        varParams will be the varParamStr column from the data base
 
-        This method uses json to convert that into a machine-readable object
+        # construct a registry of all of the variability models
+        # available to the InstanceCatalog
+        if not hasattr(self, '_methodRegistry'):
+            self._methodRegistry = {}
+            for methodname in dir(self):
+                method=getattr(self, methodname)
+                if hasattr(method, '_registryKey'):
+                    if method._registryKey not in self._methodRegistry:
+                        self._methodRegistry[method._registryKey] = method
+
+        if self.variabilityInitialized == False:
+            self.initializeVariability(doCache=True)
 
-        it uses the varMethodName to select the correct variability method from the
-        dict self._methodRegistry
+        # A numpy array of magnitude offsets.  Each row is
+        # an LSST band in ugrizy order.  Each columns is an
+        # astrophysical object from the CatSim database.
+        deltaMag = numpy.zeros((6, len(varParams_arr)))
+
+        # When the InstanceCatalog calls all of its getters
+        # with an empty chunk to check column dependencies,
+        # call all of the variability models in the
+        # _methodRegistry to make sure that all of the column
+        # dependencies of the variability models are detected.
+        if len(varParams_arr) == 0:
+            for method_name in self._methodRegistry:
+                self._methodRegistry[method_name]([],{},0)
+
+        # Keep a list of all of the specific variability models
+        # that need to be called.  There is one entry for each
+        # astrophysical object in the CatSim database.
```

Comment:
No.  It is an entry for each astrophysical object.  We ultimately end up running `np.where` on `method_name_arr` to figure out which objects get passed through which methods.  I will try to make the comment clearer.